src/herder/TransactionQueue.cpp

```diff
@@ -62,7 +66,14 @@ canReplaceByFee(TransactionFrameBasePtr tx, TransactionFrameBasePtr oldTx)
     // by INT64_MAX, while number of operations and FEE_MULTIPLIER are small.
     auto v1 = bigMultiply(newFee, oldNumOps);
     auto v2 = bigMultiply(oldFee, newNumOps);
-    return v1 >= TransactionQueue::FEE_MULTIPLIER * v2;
+    auto minFeeN = v2 * TransactionQueue::FEE_MULTIPLIER;
+    bool res = v1 >= minFeeN;
+    if (!res)
+    {
+        // this will not throw
+        minFee = bigDivide(minFeeN, int64_t(oldNumOps), Rounding::ROUND_UP);
```

Comment:
I assume you are relying on the fact that the total amount of native asset is bounded at about 50B? I think with that constraint this can't overflow.

I think it would probably be clearer and safer to use the non-throwing variant and just explicitly handle capping the `minFee` at `INT64_MAX` if it would have overflowed. This way we don't have to re-check this math if we change the fee-multiplier, or the max number of operations, or anything else.

Reply:
sure, fixed